src/Workspaces/Core/Portable/Remote/RemoteHostClient.cs

```diff
@@ -115,5 +115,57 @@ public void Dispose()
                 PinnedScope.Dispose();
             }
         }
+
+        public class NoOpClient : RemoteHostClient
+        {
+            public NoOpClient(Workspace workspace) :
+                base(workspace)
+            {
+            }
+
+            protected override Task<Session> CreateServiceSessionAsync(
+                string serviceName, PinnedRemotableDataScope snapshot, object callbackTarget, CancellationToken cancellationToken)
+            {
+                return Task.FromResult<Session>(new NoOpSession(snapshot, cancellationToken));
+            }
+
+            protected override void OnConnected()
+            {
+                // do nothing
+            }
+
+            protected override void OnDisconnected()
+            {
+                // do nothing
+            }
+
+            private class NoOpSession : Session
+            {
+                public NoOpSession(PinnedRemotableDataScope scope, CancellationToken cancellationToken) :
+                    base(scope, cancellationToken)
+                {
+                }
+
+                public override Task InvokeAsync(string targetName, params object[] arguments)
+                {
+                    return SpecializedTasks.EmptyTask;
+                }
+
+                public override Task<T> InvokeAsync<T>(string targetName, params object[] arguments)
+                {
+                    return SpecializedTasks.Default<T>();
```

Comment:
I tested 3 things I know that is enabled in RTW by default. FSA, LUT, Code lens. and made one which doesn't deal with null correctly (basically crashing) to deal with it.

shiproom or @MattGertz will decide which route we will go. once we choose one, I will probably remove all others and all features that need to run on OOP need to know how to handle OOP killed case.